### ISSUE TITLE
CI: Test our *2john tools

### ISSUE THREAD

**Issue by solardiz** [openwall/john#5433]
We need to start automatic testing of our `*2john` tools, perhaps on GitHub Actions using inputs from our https://github.com/openwall/john-samples repo. Most of these tools are Python scripts and some have dependencies, so this task might not be completed in one step. It's OK to start with testing our C compiled programs, then proceed to add more.

Also, maybe some of this (not requiring large input files) should become part of our `make check` (and individual tests automatically skipped when dependencies are not satisfied? so that `make check` remains usable to test `john` itself everywhere it currently is).

**Comment by ukasz** [comment]
Do we use self hosted github runners?

**Comment by solardiz** [comment]
> Do we use self hosted github runners?

No, we currently run CI jobs in GitHub and CircleCI infrastructure. I see no need for self-hosted for this task - testing `*2john` tools. (We could reasonably use self-hosted for testing on GPUs and for performance regression testing, but that's not part of this issue.)
